### PR TITLE
Support indicator types in packs

### PR DESCRIPTION
**Pull request by liorblob**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24239

## Description
Replace `Misc` with `IndicatorTypes`.

### CODE REVIEW

File: demisto_sdk/commands/common/update_id_set.py [hunk 1]
```diff
@@ -513,6 +515,26 @@ def process_indicator_fields(file_path, print_logs):
     return res
 
 
+def process_indicator_types(file_path, print_logs):
```
Comment (Itay4) [inline demisto_sdk/commands/common/update_id_set.py:518]: add type hints

File: demisto_sdk/commands/common/update_id_set.py [hunk 2]
```diff
@@ -513,6 +515,26 @@ def process_indicator_fields(file_path, print_logs):
...
+    Process a indicator types JSON file
+    Args:
+        file_path: The file path from indicator type folder
+        print_logs: Whether to print logs to stdout
+
+    Returns:
+        a list of indicator type data.
+    """
+    res = []
+    if (checked_type(file_path, [INDICATOR_TYPES_REGEX, PACKS_INDICATOR_TYPES_REGEX]) and
```
Comment (Itay4) [inline demisto_sdk/commands/common/update_id_set.py:529]:
```suggestion
    if checked_type(file_path, [INDICATOR_TYPES_REGEX, PACKS_INDICATOR_TYPES_REGEX] and
```

File: demisto_sdk/commands/common/constants.py [hunk 3]
```diff
@@ -586,10 +595,10 @@ def found_hidden_param(parameter_name):
 REPORT_REGEX = r'{}{}.*report-.*\.json$'.format(CAN_START_WITH_DOT_SLASH, REPORTS_DIR)
 REPORT_CHANGELOG_REGEX = r'{}{}.*_CHANGELOG.md$'.format(CAN_START_WITH_DOT_SLASH, REPORTS_DIR)
 
-MISC_REGEX = r'{}{}.*reputations\.json$'.format(CAN_START_WITH_DOT_SLASH, MISC_DIR)
-MISC_REPUTATIONS_REGEX = r'{}{}.reputations.json$'.format(CAN_START_WITH_DOT_SLASH, MISC_DIR)
-REPUTATION_REGEX = r'{}{}/reputation-.*\.json$'.format(CAN_START_WITH_DOT_SLASH, MISC_DIR)
-REPUTATION_CHANGELOG_REGEX = r'{}{}/*_CHANGELOG.md$'.format(CAN_START_WITH_DOT_SLASH, MISC_DIR)
+INDICATOR_TYPES_REGEX = r'{}{}/reputation-.*\.json$'.format(CAN_START_WITH_DOT_SLASH, INDICATOR_TYPES_DIR)
+INDICATOR_TYPES_CHANGELOG_REGEX = r'{}{}/*_CHANGELOG.md$'.format(CAN_START_WITH_DOT_SLASH, INDICATOR_TYPES_DIR)
+INDICATOR_TYPES_REPUTATIONS_REGEX = r'{}{}.reputations.json$'.format(CAN_START_WITH_DOT_SLASH, INDICATOR_TYPES_DIR)
```
Comment (yuvalbenshalom) [inline demisto_sdk/commands/common/constants.py:600]: @IkaDemisto do you think we should allow "reputations" only in the none supported pack path?

File: demisto_sdk/tests/constants_test.py [hunk 4]
```diff
@@ -49,7 +49,7 @@
 INVALID_MULTI_LINE_1_CHANGELOG_PATH = f"{GIT_ROOT}/demisto_sdk/tests/test_files/invalid-multi-line_1_CHANGELOG.md"
 INVALID_MULTI_LINE_2_CHANGELOG_PATH = f"{GIT_ROOT}/demisto_sdk/tests/test_files/invalid-multi-line_2_CHANGELOG.md"
 LAYOUT_TARGET = "./Layouts/layout-mock.json"
-REPUTATION_TARGET = "./Misc/reputations.json"
+INDICATOR_TYPE_TARGET = "IndicatorTypes/reputations.json"
```
Comment (yuvalbenshalom) [inline demisto_sdk/tests/constants_test.py:52]: I don't think that it it is good that our example of `INDICATOR_TYPE_TARGET` is the deprecated exception `reputations.json`

**Comment by Itay4** [comment]
@liorblob update changelog please

File: .flake8 [hunk 5]
```diff
@@ -8,4 +8,4 @@
 # E999 Invalid Syntax
 ignore = W293, W504, F405, F403
 max-line-length = 160
-max-complexity = 30
+max-complexity = 35
```
[the remark this answers is not in the document]
Comment (liorblob) [reply .flake8:11]: flake8 fails otherwise...

[the remark this answers is not in the document]
Comment (Itay4) [reply .flake8:11]: yea i understand, but question is what it fails on
i prefer we ignore it in that specific place

**Comment by liorblob** [comment]
@Itay4 not sure about the tests... I was trying to look at content creator tests, we don't have tests for directories. Some tests in validation and constants cover some of it, but not everything

File: .flake8 (hunk 5 above)
[the remark this answers is not in the document]
Comment (liorblob) [reply .flake8:11]: it's the id set creation function

[the remark this answers is not in the document]
Comment (liorblob) [reply .flake8:11]: I will ignore it there

File: demisto_sdk/commands/common/update_id_set.py (hunk 2 above)
Comment (liorblob) [reply demisto_sdk/commands/common/update_id_set.py:529]: the parenthesis are for the line break

**Pull request merged**